Gems/ROS2/Code/Source/Lidar/ROS2Lidar2DSensorComponent.cpp

```diff
@@ -39,186 +37,76 @@ namespace ROS2
-        }
-
-        if (m_lidarConfiguration.m_lidarSystemFeatures & LidarSystemFeatures::EntityExclusion)
-        {
-            LidarRaycasterRequestBus::Event(
-                m_lidarRaycasterId, &LidarRaycasterRequestBus::Events::ExcludeEntities, m_lidarConfiguration.m_excludedEntities);
-        }
-
-        if (m_lidarConfiguration.m_lidarSystemFeatures & LidarSystemFeatures::MaxRangePoints)
-        {
-            LidarRaycasterRequestBus::Event(
-                m_lidarRaycasterId,
-                &LidarRaycasterRequestBus::Events::ConfigureMaxRangePointAddition,
-                m_lidarConfiguration.m_addPointsAtMax);
-        }
-    }
-
     ROS2Lidar2DSensorComponent::ROS2Lidar2DSensorComponent()
+        : m_lidarCore(LidarTemplateUtils::Get2DModels())
     {
         TopicConfiguration ls;
-        AZStd::string type = Internal::kLaserScanType;
+        AZStd::string type = kLaserScanType;
         ls.m_type = type;
         ls.m_topic = "ls";
-        m_sensorConfiguration.m_frequency = 10;
+        m_sensorConfiguration.m_frequency = 10.f;
         m_sensorConfiguration.m_publishersConfigurations.insert(AZStd::make_pair(type, ls));
     }
-
     ROS2Lidar2DSensorComponent::ROS2Lidar2DSensorComponent(
-        const SensorConfiguration& sensorConfiguration, const Lidar2DSensorConfiguration& lidarConfiguration)
-        : m_lidarConfiguration(lidarConfiguration)
+        const SensorConfiguration& sensorConfiguration, const LidarSensorConfiguration& lidarConfiguration)
+        : m_lidarCore(lidarConfiguration)
     {
         m_sensorConfiguration = sensorConfiguration;
     }
 
     void ROS2Lidar2DSensorComponent::Visualize()
     {
-        if (m_visualizationPoints.empty())
-        {
-            return;
-        }
-
-        if (m_drawQueue)
-        {
-            const uint8_t pixelSize = 2;
-            AZ::RPI::AuxGeomDraw::AuxGeomDynamicDrawArguments drawArgs;
-            drawArgs.m_verts = m_visualizationPoints.data();
-            drawArgs.m_vertCount = m_visualizationPoints.size();
-            drawArgs.m_colors = &AZ::Colors::Red;
-            drawArgs.m_colorCount = 1;
-            drawArgs.m_opacityType = AZ::RPI::AuxGeomDraw::OpacityType::Opaque;
-            drawArgs.m_size = pixelSize;
-            m_drawQueue->DrawPoints(drawArgs);
-        }
+        m_lidarCore.VisualizeResults();
     }
 
     void ROS2Lidar2DSensorComponent::Activate()
     {
+        m_lidarCore.Init(GetEntityId());
+
         auto ros2Node = ROS2Interface::Get()->GetNode();
         AZ_Assert(m_sensorConfiguration.m_publishersConfigurations.size() == 1, "Invalid configuration of publishers for lidar sensor");
 
-        const TopicConfiguration& publisherConfig = m_sensorConfiguration.m_publishersConfigurations[Internal::kLaserScanType];
+        const TopicConfiguration& publisherConfig = m_sensorConfiguration.m_publishersConfigurations[kLaserScanType];
```

Comment:
Pleasse follow [Constant names](https://github.com/o3de/sig-core/blob/main/governance/Coding-Standards-and-Style-Guide.md#constant-names) section of coding rules with this constant.